src/main/java/Graph.java

```diff
@@ -1,31 +1,40 @@
 import java.util.ArrayList;
-import java.util.LinkedList;
+import java.util.HashMap;
 import java.util.List;
 
 public class Graph {
-    private LinkedList<GraphNode> nodes;
-    private List<GraphNode> pointsOfFailure = new ArrayList<>();
+    private final List<GraphNode> nodes;
+    private final List<GraphNode> pointsOfFailure = new ArrayList<>();
+    private int numSubnets = 0;
+    private final HashMap<GraphNode, Integer> subnetHashmap = new HashMap<>();
     private int time = 0;
 
-    Graph(){ }
-
-    Graph(LinkedList<GraphNode> nodes) {
+    Graph(List<GraphNode> nodes) {
         this.nodes = nodes;
+        findSPF();
+    }
+
+    public List<GraphNode> getGraph() {
+        return nodes;
+    }
+
+    public List<GraphNode> getSPF() {
+        return pointsOfFailure;
     }
 
-    public List<GraphNode> findSPF()  {
-        // Mark all the vertices as not visited
+    public HashMap<GraphNode, Integer> getSubnets() {
+
+        return subnetHashmap;
+    }
 
+    private void  findSPF()  {
```

Comment:
As discussed on our call, perhaps this method can be made smaller- though not priority